File: Shared/Resources/selectionsort.bundle/selectionsort.go
package main

import (
	"fmt"
)

func selectionSort(arr []int) {
	n := len(arr)
	for i := 0; i < n - 1; i++ {
		minIdx := i
		for j := i + 1; j < n; j++ {
			if arr[j] < arr[minIdx] {
				minIdx = j
			}
		}
		arr[minIdx], arr[i] = arr[i], arr[minIdx]
	}
}

func main() {
  items := []int{35, 95, 74, 71, 72, 30, 96, 53, 9, 0}
  selectionSort(items)
  fmt.Println(items)
}

File: Shared/Resources/selectionsort.bundle/selectionsort.rb
def selectionSort(arr)
  n = arr.length()
  i = 0
  while i < n - 1
    minIdx = i
    j = i + 1
    while j < n
      if arr[j] < arr[minIdx]
        minIdx = j
      end
      j += 1
    end
    arr[minIdx], arr[i] = arr[i], arr[minIdx]
    i += 1
  end
end

items = [35, 95, 74, 71, 72, 30, 96, 53, 9, 0]
selectionSort(items)
p items
